automake/post/rules/makedirs.am

```diff
@@ -31,42 +31,102 @@
 #        (all _OBJECTS) defined by a Makefile.am to depend upon an invocation
 #        of `$(MAKE) -C <DIR>` in that directory.
 #
+
+
+# same as ifdef MAKEDIRS || MAKEDIR_TARGETS
+ifneq ($(strip $(MAKEDIRS)$(MAKEDIR_TARGETS)),)
+
+# check to see if makedirs_id has been set for this invocation
+ifneq ($(origin makedirs_id),environment)
+
+  # the parent of this shell is the "root" make invocation
+  override makedirs_id:=makedirs_$(shell echo $$PPID)
+
+  # makes makedirs_id an environment var
+  export makedirs_id
+
+endif
+
+endif # ifneq ($(strip $(MAKEDIRS)$(MAKEDIR_TARGETS)),)
+
+#
+# MAKEDIRS section
+#
 ifdef MAKEDIRS
 
-.PHONY: makedirs force
+.PHONY: makedirs
+
+ifeq ($(MAKECMDGOALS),$(filter-out clean distclean,$(MAKECMDGOALS)))
+# trick make into building makedirs first by forcing it to try to make an
```

Comment:
Is there any reason to not leave it to the makefile developer to specify this if they want it / as they need it?

Reply:
A user of MAKEDIRS is asking "please run make in these directories because I have implicit dependencies satisfied by running $MAKE those directories".